cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +795,25 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+int32_t gdv_fn_instr_utf8(const char* string, int32_t string_len, const char* substring,
+                          int32_t substring_len) {
+  if (substring_len == 0) {
+    return 1;
+  }
+
+  if (string_len < substring_len) {
```

Comment:
What happens if string_len == substring_len? Looks like we return 0 always - the for loop goes from 0 to 0

Can you fix this case?

Also, can this function be moved to precompiled and use FORCE_INLINE?

Reply:
Fixed this if clause now, it will handle case when string_len == substring_len and when string_len < substring_len